### PR TITLE
refact(RHINENG-15309): Refactor filtering logic related to custom staleness

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-15309](https://issues.redhat.com/browse/RHINENG-15309).
The purpose of this PR is to simplify some things, but it also resolves an inconsistency with staleness_to_conditions, which is needed to unblock #2181.

- Move ALL_STALENESS_STATES to config.py
- Consolidate stale_timestamp filter functions
- Move staleness_to_conditions and update_query_for_owner_id to db_filters.py, as they both generate DB filters
- Resolve the inconsistency in staleness_to_conditions timestamp filtering functions. Some required host_type, and some did not.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [x] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/filtering/db_filters.py [hunk 1]
```diff
@@ -156,11 +166,11 @@ def _staleness_filter(
     staleness_obj = serialize_staleness_to_dict(get_staleness_obj(identity))
     staleness_conditions = []
     for host_type in host_type_filter:
-        conditions = or_(*staleness_to_conditions(staleness_obj, staleness, host_type, _stale_timestamp_filter))
+        conditions = or_(*staleness_to_conditions(staleness_obj, staleness, host_type, stale_timestamp_filter, True))
```
Comment (thearifismail) [inline api/filtering/db_filters.py:169]: Won't the code be easier to read if `*` is not used with `staleness_to_conditions()` call as the number of iterable members is only 3?

Comment (kruai) [reply api/filtering/db_filters.py:169]: The `or_()` function takes 2 or more expression/clause arguments, not a tuple or list. We have to use the unpack operator (`*`) as far as I can tell, but I'm open to suggestions

File: api/filtering/db_filters.py [hunk 2]
```diff
@@ -170,6 +180,21 @@ def _staleness_filter(
     return [or_(*staleness_conditions)]
 
 
+def staleness_to_conditions(
+    staleness, staleness_states, host_type, timestamp_filter_func, omit_host_type_filter: bool = False
```
[the remark this answers is not in the document]
Comment (kruai) [reply api/filtering/db_filters.py:184]: This function is retaining the same signature as what it had when it was in `culling.py`.
Also, I don't see `staleness_type` used anywhere; can you please point me to that?

File: app/config.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 PRODUCER_ACKS = {"0": 0, "1": 1, "all": "all"}
 
 HOST_TYPES = ["edge", None]
+ALL_STALENESS_STATES = ("fresh", "stale", "stale_warning")
```
[the remark this answers is not in the document]
Comment (kruai) [reply app/config.py:15]: All I'm doing is moving this constant from `host_repository.py` to `config.py`, not changing any values. But no, we don't allow querying hosts that are in the culled state

[the remark this answers is not in the document]
Comment (thearifismail) [reply app/config.py:15]: I think it is needed so I filed [RHINENG-15328](https://issues.redhat.com/browse/RHINENG-15328).

File: api/filtering/db_filters.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (thearifismail) [reply api/filtering/db_filters.py:184]: oh it was `staleness_types` on line 206, 207, and 228 in `lib/host_repository.py`

File: api/filtering/db_filters.py (hunk 1 above)
Comment (thearifismail) [reply api/filtering/db_filters.py:169]: Since the code works.  I am good.

**Review by thearifismail — APPROVED** [review]
LGTM
